### PR TITLE
feat: implement no bargain event

### PR DESCRIPTION
**Pull request by jgordijo**
image

### CODE REVIEW

**Review by adbronico — DISMISSED** [review]
This looks good to me. Wait on Veld to do a final review, though.

File: test/server/cards/02_SHD/events/NoBargain.spec.ts
Comment (AMMayberry1) [inline test/server/cards/02_SHD/events/NoBargain.spec.ts:16]: If you could do me a favor - the discard system is new, we'd like to add a few extra tests on the first few cards that use it. Could you add one more case where the opponent has no cards in hand and confirm that the draw still happens?

Comment (jgordijo) [reply test/server/cards/02_SHD/events/NoBargain.spec.ts:16]: @AMMayberry1 Done!

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**
